### PR TITLE
Add lorem command

### LINKED ISSUE #1341: Generating Random Lorem Text
**New Command Idea**

- Create a new command, say "s!lorem" which will generate a random lorem ipsum text.

Here is a module for generating lorem ipsum text - https://github.com/knicklabs/lorem-ipsum.js

### PR DESCRIPTION
**Pull request by AsianCat54x**
Add a new command which generates random lorem text

Closes #1341 

Edit: Not again! All my previous commit came in the way :(

### CODE REVIEW

File: src/commands/misc/lorem.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const { SwitchbladeEmbed, Command } = require('../../')
+const LoremIpsum = require('lorem-ipsum').LoremIpsum
```
Comment (almeidx) [inline src/commands/misc/lorem.js:2]:
```suggestion
const { LoremIpsum } = require('lorem-ipsum')
```

File: src/commands/misc/lorem.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+module.exports = class Lorem extends Command {
+  constructor (client) {
+    super({
+      name: 'lorem',
+      parameters: [{
+        type: 'number',
+        full: true,
+        missingError: 'commands:lorem.notFound',
```
Comment (almeidx) [inline src/commands/misc/lorem.js:11]: This string doesn't exist. And `notFound` sounds misleading as well

File: src/locales/en-US/commands.json [hunk 3]
```diff
@@ -672,6 +672,10 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "lorem": {
+    "commandDescription": "Generates a random lorem text with speciifed number of paragraphs",
+    "commandUsage": "<no of paragraphs>"
```
Comment (almeidx) [inline src/locales/en-US/commands.json:677]:
```suggestion
    "commandUsage": "<number of paragraphs>"
```

File: src/locales/en-US/commands.json [hunk 4]
```diff
@@ -672,6 +672,10 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "lorem": {
+    "commandDescription": "Generates a random lorem text with speciifed number of paragraphs",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:676]:
```suggestion
    "commandDescription": "Generates a random lorem ipsum text with the specified amount of paragraphs.",
```

File: src/commands/misc/lorem.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      }]
+    },
+    client)
+
+    this.lorem = new LoremIpsum()
+  }
+
+  async run ({ channel }, paragraphs) {
+    const embed = new SwitchbladeEmbed()
+      .setDescription(this.lorem.generateParagraphs(paragraphs))
```
Comment (almeidx) [inline src/commands/misc/lorem.js:22]: This will throw an error if the length is over 4096.

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/misc/lorem.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    this.lorem = new LoremIpsum()
+  }
+
+  async run ({ channel , t}, paragraphs) {
+    if(paragraphs > 4096 || paragraphs < 1){
+      throw new CommandError(t("commands:lorem.exceededNumber"))
+    }
+    const embed = new SwitchbladeEmbed()
+      .setDescription(this.lorem.generateParagraphs(paragraphs))
```
Comment (almeidx) [inline src/commands/misc/lorem.js:25]: The number of paragraphs does not represent the length of the final string.

Comment (AsianCat54x) [reply src/commands/misc/lorem.js:25]: Ate you telling that the maximum length of the string should be less than 4096?

Comment (almeidx) [reply src/commands/misc/lorem.js:25]: Yes

Comment (AsianCat54x) [reply src/commands/misc/lorem.js]: Oh, done :)

File: src/commands/misc/lorem.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+module.exports = class Lorem extends Command {
+  constructor (client) {
+    super({
+      name: 'lorem',
+      parameters: [{
+        type: 'number',
+        full: true,
+        missingError: 'commands:lorem.noNumber',
+        required: true
```
Comment (almeidx) [inline src/commands/misc/lorem.js:12]: Dont think this should be required. Perhaps default it to 1 or 2 instead?

Comment (AsianCat54x) [reply src/commands/misc/lorem.js:12]: Two words?

Comment (AsianCat54x) [reply src/commands/misc/lorem.js]: Oh, okay, done :)

File: src/commands/misc/lorem.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+module.exports = class Lorem extends Command {
+  constructor (client) {
+    super({
+      name: 'lorem',
+      parameters: [{
+        type: 'number',
+        full: true,
+        missingError: 'commands:lorem.noNumber',
+        required: false,
+        default: 2
```
Comment (almeidx) [inline src/commands/misc/lorem.js:13]: `default` is not an option.

```suggestion
        type: 'number',
        missingError: 'commands:lorem.noNumber',
        required: false,
        min: 1,
        max: 20,
```

PS. I don't know how many words each paragraph will have, so you may increase/decrease the max if you want

Comment (AsianCat54x) [reply src/commands/misc/lorem.js:13]: What, default is not an option! But I think I saw it somewhere

Comment (AsianCat54x) [reply src/commands/misc/lorem.js]: Probably my mistake. I thought that default is an option :(

**Comment by almeidx** [comment]
Just to clarify: I didn't say to change the parameter to words instead of paragraphs. What I was saying is that if the generated had more than 4096 characters then it would throw an error.

**Comment by AsianCat54x** [comment]
Then I must have misunderstood. So I should check if the return string has more than 4096, if yes, then throw a command error?

File: src/commands/misc/lorem.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+    this.lorem = new LoremIpsum()
+  }
+
+  async run ({ channel, t }, paragraphs) {
+    if(paragraphs === undefined) {
+      paragraphs = 2
+    }
+
+    const loremString = this.lorem.generateParagraphs(paragraphs)
```
Comment (metehus) [inline src/commands/misc/lorem.js:27]:
```js
async run ({ channel, t }, paragraphs) {
  const loremString = this.lorem.generateParagraphs(paragraphs || 2)
```

Comment (AsianCat54x) [reply src/commands/misc/lorem.js:27]: Should have thought about it! One-liner :)

File: src/commands/misc/lorem.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+      }]
+    },
+    client)
+
+    this.lorem = new LoremIpsum()
+  }
+
+  async run ({ channel, t }, paragraphs) {
+    const loremString = this.lorem.generateParagraphs(paragraphs || 2)
+
```
Comment (metehus) [inline src/commands/misc/lorem.js:24]:
```suggestion
    let loremString = this.lorem.generateParagraphs(paragraphs || 2)
    if (loremString.length >= 4096) {
      loremString = loremString.slice(0, 4095) + '…'
    }

```

File: src/locales/en-US/commands.json [hunk 11]
```diff
@@ -672,6 +672,12 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "lorem": {
+    "commandDescription": "Generates a random lorem ipsum text with the specified amount of words.",
+    "commandUsage": "<number of words>",
+    "noNumber": "Number of words to generate is missing!",
+    "exceededNumber": "Number of words should be in a range of 1 to 4096"
```
Comment (metehus) [inline src/locales/en-US/commands.json:679]: This string is not being used
```suggestion
```

**Review by metehus — CHANGES_REQUESTED** [review]
Besides these suggestions, its LGTM

**Review by fersilva16 — APPROVED** [review]
LGTM!

File: src/commands/misc/lorem.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
+const { SwitchbladeEmbed, Command } = require('../../')
+const { LoremIpsum } = require('lorem-ipsum')
+
+module.exports = class Lorem extends Command {
+  constructor (client) {
+    super({
+      name: 'lorem',
+      parameters: [{
+        type: 'number',
+        full: true,
```
Comment (almeidx) [inline src/commands/misc/lorem.js:10]:
```suggestion
```

File: src/commands/misc/lorem.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+        max: 10
+      }]
+    },
+    client)
+
+    this.lorem = new LoremIpsum()
+  }
+
+  async run ({ channel, t }, paragraphs) {
+    let loremString = this.lorem.generateParagraphs(paragraphs || 2)
```
Comment (almeidx) [inline src/commands/misc/lorem.js:23]:
```suggestion
  async run ({ channel, t }, paragraphs = 2) {
    let loremString = this.lorem.generateParagraphs(paragraphs)
```

File: src/locales/en-US/commands.json [hunk 14]
```diff
@@ -672,6 +672,11 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "lorem": {
+    "commandDescription": "Generates a random lorem ipsum text with the specified amount of words.",
+    "commandUsage": "<number of words>",
+    "noNumber": "Number of words to generate is missing!"
```
Comment (almeidx) [inline src/locales/en-US/commands.json:678]:
```suggestion
    "noNumber": "You have to give me a number between 1 and 10!"
```

File: src/locales/en-US/commands.json [hunk 15]
```diff
@@ -672,6 +672,11 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "lorem": {
+    "commandDescription": "Generates a random lorem ipsum text with the specified amount of words.",
+    "commandUsage": "<number of words>",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:677]:
```suggestion
    "commandUsage": "<number of paragraphs>",
```

File: src/locales/en-US/commands.json [hunk 16]
```diff
@@ -672,6 +672,11 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "lorem": {
+    "commandDescription": "Generates a random lorem ipsum text with the specified amount of words.",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:676]:
```suggestion
    "commandDescription": "Generates a random lorem ipsum text with the specified amount of paragraphs.",
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/misc/lorem.js (hunk 13 above)
Comment (almeidx) [reply src/commands/misc/lorem.js:22]: This was not resolved

Comment (AsianCat54x) [reply src/commands/misc/lorem.js:22]: Why so? I changed it :D

Comment (almeidx) [reply src/commands/misc/lorem.js:22]: You did not

Comment (AsianCat54x) [reply src/commands/misc/lorem.js]: Oh I guess it got undo while I was stuck on rebasing :(

Comment (AsianCat54x) [reply src/commands/misc/lorem.js]: I have done the changes, you can now review ;)

**Review by almeidx — APPROVED** [review]

**Review by metehus — APPROVED** [review]
Good

**Review by Doges — APPROVED** [review]
LGTM

**Comment by AsianCat54x** [comment]
Thanks!

**Pull request merged**